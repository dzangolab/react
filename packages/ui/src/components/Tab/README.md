# TabbedPanel component

## Installation

### npm

```
npm install @dzangolab/react-ui
```

### yarn

```
yarn add @dzangolab/react-ui
```

### pnpm

```
pnpm add @dzangolab/react-ui
```

## Usage

```
import React, { ReactNode } from "react";

import { SubPane, TabbedPanel } from "@dzangolab/react-ui";


const Page: React.FC<Properties> = ({ children }) => {
  return (
    <TabbedPanel>
      <SubPane
        icon="icon1.jpg"
        title="Topic 1">
        Pane 1 content
      </SubPane>
      <SubPane
        icon="icon2.jpg"
        title="Topic 2">
        Pane 2content
      </SubPane>
    </TabbedPanel>
  );
};

export default Page
```

## Props

### Accordion

| Prop      | Type                       | Description                                      | Default      |
|-----------|----------------------------|--------------------------------------------------|--------------|
| position  | `top` \| `left` \| `bottom` \| `right` | The position of the tabs relative to the tabbed pane. | `top`

### SubPane

| Prop  | Type     | Description                                     | Default      |
|-------|----------|-------------------------------------------------|--------------|
| icon  | `string` | The icon to display in the subpane's title area | None         |
| title | `string` | The text to display in the subpan'es title area | None         |
