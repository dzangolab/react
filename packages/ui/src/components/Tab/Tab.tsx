import React, { ReactNode } from "react";

type Properties = {
  children: ReactNode;
  title: string;
};

const Tab: React.FC<Properties> = ({ children }) => {
  return <div className="tab">{children}</div>;
};

export default Tab;
