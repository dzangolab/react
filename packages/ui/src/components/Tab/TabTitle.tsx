import React from "react";

type Properties = {
  title: string;
  eventKey: number;
  selected: number;
  setSelectedTab: (index: number) => void;
};

const TabTitle: React.FC<Properties> = ({
  title,
  selected,
  setSelectedTab,
  eventKey,
}) => {
  return (
    <li>
      <button
        className={`tab-title ${selected === eventKey ? "active" : ""}`}
        onClick={() => setSelectedTab(eventKey)}
      >
        {title}
      </button>
    </li>
  );
};

export default TabTitle;
