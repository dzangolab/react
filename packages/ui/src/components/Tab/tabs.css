.tab-list {
  align-items: flex-start;
  display: flex;
  flex-direction: row;
  gap: 1rem;
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.tab-list button {
  background-color: hsl(0deg 0% 90%);
  color: hsl(0deg 0% 13%);
  border: 1px solid hsl(0deg 0% 13%);
  padding: 0.5rem 1rem;
  cursor: pointer;
  border-radius: 4px;
}

.tab-list .tab-title.active {
  background-color: #fff;
  color: #1c4786;
}

.tab-list .tab-title:focus {
  outline: 1px solid #1c4786;
  outline-offset: 1px;
}

.tab {
  padding: 1rem 0;
}
