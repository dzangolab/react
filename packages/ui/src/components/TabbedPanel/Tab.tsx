import React from "react";

type Properties = {
  title: string;
  icon: string;
  index: number;
  isActive: boolean;
  handleClick: (index: number) => void;
};

const Tab: React.FC<Properties> = ({
  title,
  icon,
  index,
  isActive,
  handleClick,
}) => {
  return (
    <li
      className={`tab ${isActive ? "active" : ""}`}
      onClick={() => handleClick(index)}
    >
      {icon ? <img src={icon} className="tab-icon" alt={title} /> : null}
      {title}
    </li>
  );
};

export default Tab;
