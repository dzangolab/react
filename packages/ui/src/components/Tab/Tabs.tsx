import React, { ReactElement, useState } from "react";

import TabTitle from "./TabTitle";

import "./tabs.css";

type Properties = {
  children: ReactElement[];
};

const Tabs: React.FC<Properties> = ({ children }) => {
  const [selected, setSelected] = useState(0);

  return (
    <div className="tabs">
      <ul className="tab-list">
        {children.map((item, index) => (
          <TabTitle
            key={index}
            title={item.props.title}
            index={index}
            selected={selected}
            setSelectedTab={setSelected}
          />
        ))}
      </ul>
      {children[selected]}
    </div>
  );
};

export default Tabs;
