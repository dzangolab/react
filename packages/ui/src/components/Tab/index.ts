import Tab from "./Tab";
import Tabs from "./Tabs";
import TabTitle from "./TabTitle";

export { Tab, Tabs, TabTitle };
