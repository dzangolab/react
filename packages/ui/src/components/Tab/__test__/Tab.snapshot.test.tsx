import React from "react";
import renderer, { create } from "react-test-renderer";
import { expect, test } from "vitest";

import Tabs from "../Tabs";
import Tab from "../Tab";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as renderer.ReactTestRendererJSON;
}

test("Component matches snapshot", () => {
  const component = create(
    <Tabs>
      <Tab title="Tab one">
        <p>Some text</p>
      </Tab>
      <Tab title="Tab two">
        <p>Some text</p>
      </Tab>
    </Tabs>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
