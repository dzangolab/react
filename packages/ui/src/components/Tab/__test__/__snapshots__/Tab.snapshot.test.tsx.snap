// Vitest Snapshot v1

exports[`Component matches snapshot 1`] = `
<div
  className="tab"
>
  <p>
    tab body
  </p>
</div>
`;
